x-pack/plugins/security_solution/public/management/components/endpoint_execute_action/endpoint_execute_action.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { memo } from 'react';
+import {
+  EuiAccordion,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiText,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import {
+  ResponseActionFileDownloadLink,
+  type ResponseActionFileDownloadLinkProps,
+} from '../response_action_file_download_link';
+import type { ResponseActionExecuteOutputContent } from '../../../../common/endpoint/types';
+
+const EXECUTE_FILE_LINK_TITLE = i18n.translate(
+  'xpack.securitySolution.responseActionExecuteDownloadLink.downloadButtonLabel',
+  { defaultMessage: 'Click here to download full output' }
+);
+
+const ACCORDION_BUTTON_TEXT = Object.freeze({
+  output: {
+    regular: i18n.translate(
+      'xpack.securitySolution.responseActionExecuteAccordion.outputButtonTextRegular',
+      {
+        defaultMessage: 'Execution output',
+      }
+    ),
+    truncated: i18n.translate(
+      'xpack.securitySolution.responseActionExecuteAccordion.outputButtonTextTruncated',
+      {
+        defaultMessage: 'Execution output (truncated)',
+      }
+    ),
+  },
+  error: {
+    regular: i18n.translate(
+      'xpack.securitySolution.responseActionExecuteAccordion.errorButtonTextRegular',
+      {
+        defaultMessage: 'Execution error',
+      }
+    ),
+    truncated: i18n.translate(
+      'xpack.securitySolution.responseActionExecuteAccordion.errorButtonTextTruncated',
+      {
+        defaultMessage: 'Execution error (truncated)',
+      }
+    ),
+  },
+});
+interface ExecuteActionOutputProps {
+  content: string;
+  id: string;
```

Comment:
Do you really need to pass in the ID? could you just generated it internally in this component? or do you have a need for that ID to be available outside of this component?

Reply:
Yeah, I can generate the id inside the according component actually. Will do.

Reply:
997bde19c8da4e7798793f8fcd7041902412a6f1